storage/src/main/java/tech/pegasys/teku/storage/server/kvstore/dataaccess/V4FinalizedKvStoreDao.java

```diff
@@ -116,6 +122,54 @@ public void close() throws Exception {
     return db.get(schema.getColumnNonCanonicalBlocksByRoot(), root);
   }
 
+  @Override
+  public void ingest(
+      final KvStoreFinalizedDao finalizedDao, final int batchSize, final Consumer<String> logger) {
+    Preconditions.checkArgument(batchSize > 1, "Batch size must be greater than 1 element");
+    Preconditions.checkArgument(finalizedDao instanceof V4FinalizedKvStoreDao);
+
+    final Map<String, KvStoreVariable<?>> newVariables = schema.getVariableMap();
+    if (newVariables.size() > 0) {
+      final Map<String, KvStoreVariable<?>> oldVariables =
+          ((V4FinalizedKvStoreDao) finalizedDao).schema.getVariableMap();
+      try (V4FinalizedUpdater updater = new V4FinalizedUpdater(db, schema, UInt64.ONE)) {
+        for (String key : newVariables.keySet()) {
+          logger.accept(String.format("Copy variable %s", key));
+          finalizedDao
+              .getRawVariable(oldVariables.get(key))
+              .ifPresent(value -> updater.transaction.putRaw(newVariables.get(key), value));
+        }
+        updater.commit();
+      }
+    }
+    final Map<String, KvStoreColumn<?, ?>> newColumns = schema.getColumnMap();
+    if (newColumns.size() > 0) {
+      final Map<String, KvStoreColumn<?, ?>> oldColumns =
+          ((V4FinalizedKvStoreDao) finalizedDao).schema.getColumnMap();
```

Comment:
nit: I'd cast to the specific type once a the top of the method and store in a variable.